File: backend/app/api/schedules.py
"""日程接口：列表 / 新建 / 删除。

依据 docs/figma-design-spec.md §4 日程列表页：
- GET    /api/schedules           列出当前用户的全部日程（按日期升序）
- POST   /api/schedules           新建日程（标题/日期/紧急程度/关联案件）
- DELETE /api/schedules/{id}      删除日程
"""
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.core.deps import get_current_user, get_db
from app.models.schedule import Schedule
from app.models.user import User
from app.schemas.schedule import ScheduleCreate, ScheduleOut

router = APIRouter(prefix="/api/schedules", tags=["schedules"])


@router.get("", response_model=list[ScheduleOut])
def list_schedules(
    user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    """列出当前用户的全部日程，按日期升序。"""
    return (
        db.query(Schedule)
        .filter(Schedule.user_id == user.id)
        .order_by(Schedule.date.asc())
        .all()
    )


@router.post("", response_model=ScheduleOut)
def create_schedule(
    payload: ScheduleCreate,
    user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    """新建日程。"""
    schedule = Schedule(
        user_id=user.id,
        title=payload.title.strip(),
        date=payload.date,
        level=payload.level,
        remind_advance=payload.remind_advance,
        case_name=payload.case_name,
        case_id=payload.case_id,
    )
    db.add(schedule)
    db.commit()
    db.refresh(schedule)
    return schedule


@router.delete("/{schedule_id}")
def delete_schedule(
    schedule_id: int,
    user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    """删除日程（仅本人）。"""
    schedule = db.get(Schedule, schedule_id)
    if not schedule or schedule.user_id != user.id:
        raise HTTPException(status_code=404, detail="日程不存在")
    db.delete(schedule)
    db.commit()
    return {"ok": True}

File: frontend/src/api/schedules.ts
import request from './request';

/** 日程紧急程度：urgent 紧急 / normal 一般 / meeting 会议 */
export type ScheduleLevel = 'urgent' | 'normal' | 'meeting';

/** 日程项（对应后端 ScheduleOut） */
export interface ScheduleItem {
  id: number;
  user_id: number;
  title: string;
  date: string; // YYYY-MM-DD
  level: ScheduleLevel;
  /** 提前提醒天数：0=当天，n=提前 n 天，null=不提醒 */
  remind_advance?: number | null;
  case_name?: string | null;
  case_id?: number | null;
  created_at: string;
}

export interface ScheduleCreatePayload {
  title: string;
  date: string; // YYYY-MM-DD
  level: ScheduleLevel;
  remind_advance?: number | null;
  case_name?: string;
  case_id?: number;
}

export const schedulesApi = {
  list: () => request.get<ScheduleItem[]>('/schedules').then((r) => r.data),

  create: (payload: ScheduleCreatePayload) =>
    request.post<ScheduleItem>('/schedules', payload).then((r) => r.data),

  remove: (id: number) =>
    request.delete<{ ok: true }>(`/schedules/${id}`).then((r) => r.data),
};
